buildenv/jenkins/ARTIFACTORY.md

```diff
@@ -0,0 +1,200 @@
+The template for the reverse proxy was taken from https://www.jfrog.com/confluence/display/RTF/Configuring+NGINX and there were some changes to it. 
+This is the copy that is currently being used 
+```
+## server configuration
+server {
+
+    server_name 140-211-168-230-openstack.osuosl.org;
+    if ($http_x_forwarded_proto = '') {
+        set $http_x_forwarded_proto  $scheme;
+    }
+
+    ## Application specific logs
+    ## access_log /var/log/nginx/artifactory.log timing;
+    ## error_log /var/log/nginx/artifactory-error.log;
+    rewrite ^/$ /artifactory/webapp/ redirect;
+    rewrite ^/artifactory/?(/webapp)?$ /artifactory/webapp/ redirect;
+    chunked_transfer_encoding on;
+    client_max_body_size 0;
+    location /artifactory/ {
+    proxy_read_timeout  900;
+    proxy_pass_header   Server;
+    proxy_cookie_path   ~*^/.* /;
+    if ( $request_uri ~ ^/artifactory/(.*)$ ) {
+        proxy_pass          http://localhost:8081/artifactory/$1;
+    }
+    proxy_pass         http://localhost:8081/artifactory/;
+    proxy_set_header    X-Artifactory-Override-Base-Url $http_x_forwarded_proto://$host:$server_port/artifactory;
+    proxy_set_header    X-Forwarded-Port  $server_port;
+    proxy_set_header    X-Forwarded-Proto $http_x_forwarded_proto;
+    proxy_set_header    Host              $http_host;
+    proxy_set_header    X-Forwarded-For   $proxy_add_x_forwarded_for;
+    }
+
+    listen [::]:443 ssl ipv6only=on; # managed by Certbot
+    listen 443 ssl; # managed by Certbot
+    ssl_certificate /etc/letsencrypt/live/140-211-168-230-openstack.osuosl.org/fullchain.pem; # managed by Certbot
+    ssl_certificate_key /etc/letsencrypt/live/140-211-168-230-openstack.osuosl.org/privkey.pem; # managed by Certbot
+    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
+    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
+
+}
+
+server {
+    if ($host = 140-211-168-230-openstack.osuosl.org) {
+        return 301 https://$host$request_uri;
+    } # managed by Certbot
+
+
+    listen 80 ;
+    listen [::]:80 ;
+
+    server_name 140-211-168-230-openstack.osuosl.org;
+    return 404; # managed by Certbot
+}
+```
+All of the lines with the `# managed by Certbot` were added when adding in the ssl certificate. 
+
+Those lines do not need to be added as they were automatically added by certbot
+
+I am getting my certificate from Let’s Encrypt and their install tool certbot. Firstly, I installed certbot:
+```
+sudo apt-get update
+sudo apt-get install software-properties-common
+sudo add-apt-repository universe
+sudo add-apt-repository ppa:certbot/certbot
+sudo apt-get update
+sudo apt-get install certbot python-certbot-nginx
+```
+Then I used the certbot automated script to add the certificates and to automatically renew the certificates: 
+```
+sudo certbot –nginx
+```
+Follow the onscreen instructions and everything should work out. 
+
+# Open the Ports
+Log into OpenStack and go to network security groups
```

Comment:
Should add openstack link
https://openpower-controller.osuosl.org/auth/login/?next=/project/